pkg/remediator/reconcile/reconciler.go

```diff
@@ -85,11 +90,21 @@ func (r *reconciler) Remediate(ctx context.Context, id core.ID, obj client.Objec
 	// Record duration, even if there's an error
 	metrics.RecordRemediateDuration(ctx, metrics.StatusTagKey(err), id.WithVersion(""), start)
 
-	// Record conflict, if there was one
-	if err != nil && err.Code() == syncerclient.ResourceConflictCode {
-		metrics.RecordResourceConflict(ctx, id.WithVersion(""), commit)
+	if err != nil {
+		switch err.Code() {
+		case syncerclient.ResourceConflictCode:
+			// Record conflict, if there was one
+			metrics.RecordResourceConflict(ctx, id.WithVersion(""), commit)
```

Comment:
Wanna do everyone a favor later and convert all the metrics record functions to just use a kind string, instead of the whole GVK, since we only record the kind? It's so silly we keep having to pass an empty version to build a GVK.

Reply:
Sure. Can I make the change after this PR?